### PR TITLE
perf: improve gas price calc

### PR DESCRIPTION
**Pull request by mattsse**
refactors gas price calc.

the previous impl looped over every tx, even though we only need `limit`.

see also:

https://github.com/ethereum/go-ethereum/blob/0004c6b229b787281760b14fb9460ffd9c2496f1/eth/gasprice/gasprice.go#L259-L271

I think we could also cache the prices per block, because this does not change, right?
and considering that the range is quite large this could improve it a lot.

wdyt @Rjected ?

### CODE REVIEW

File: crates/rpc/rpc/src/eth/gas_oracle.rs [hunk 1]
```diff
@@ -11,7 +11,7 @@ use tokio::sync::Mutex;
 use tracing::warn;
 
 /// The number of transactions sampled in a block
-pub const SAMPLE_NUMBER: u32 = 3;
+pub const SAMPLE_NUMBER: usize = 3usize;
```
Comment (rkrasiuk) [inline crates/rpc/rpc/src/eth/gas_oracle.rs:14]:
```suggestion
pub const SAMPLE_NUMBER: usize = 3_usize;
```

File: crates/rpc/rpc/src/eth/gas_oracle.rs [hunk 2]
```diff
@@ -201,51 +204,54 @@ where
...
+                .ok_or(RpcInvalidTransactionError::FeeCapTooLow)?;
+
+            prices.push(U256::from(effective_gas_tip));
+
+            if prices.len() < limit {
+                continue
+            }
+
+            // we have enough entries
+            break
```
Comment (rkrasiuk) [inline crates/rpc/rpc/src/eth/gas_oracle.rs:251]:
```suggestion
            // we have enough entries
            if prices.len() >= limit {
                break
            }
```

**Review by rkrasiuk — APPROVED** [review]
lgtm, few nits

**Comment by Rjected** [comment]
> I think we could also cache the prices per block, because this does not change, right? and considering that the range is quite large this could improve it a lot.
> 
> wdyt @Rjected ?

yep, we can definitely do this

**Review by Rjected — APPROVED** [review]
LGTM

**Pull request merged**